catalog/core/catalog-core-subscriptionstore/subscriptionstore-impl/src/test/java/org/codice/ddf/catalog/subscriptionstore/common/CachedSubscriptionTest.java

```diff
@@ -0,0 +1,158 @@
+import org.codice.ddf.catalog.subscriptionstore.internal.SubscriptionRegistrationException;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceReference;
+import org.osgi.framework.ServiceRegistration;
+
+import ddf.catalog.event.Subscription;
+
+@RunWith(MockitoJUnitRunner.class)
+public class CachedSubscriptionTest {
+    private static final String TYPE = "type";
+
+    private static final String NOT_TYPE = "not_type";
+
+    private static final String FILTER = "filter";
+
+    private static final String CALLBACK = "http://localhost:1234/test";
+
+    private static final String SUBSCRIPTION_ID_OSGI = "subscription-id";
+
+    private static final String EVENT_ENDPOINT = "event-endpoint";
+
+    @Mock
+    private BundleContext mockBundleContext;
+
+    @Mock
+    private ServiceRegistration mockRegistration;
+
+    @Mock
+    private Subscription mockSubscription;
+
+    @Mock
+    private SubscriptionFactory mockFactory;
+
+    SubscriptionMetadata metadata;
+
+    private CachedSubscription cachedSubscription;
+
+    @Before
+    public void setup() {
+        ServiceReference mockRef = mock(ServiceReference.class);
+        Bundle mockBundle = mock(Bundle.class);
+        when(mockRegistration.getReference()).thenReturn(mockRef);
+        when(mockRef.getBundle()).thenReturn(mockBundle);
+        when(mockBundle.getBundleId()).thenReturn(10L);
+
+        when(mockBundleContext.registerService(anyString(), anyObject(), anyObject())).thenReturn(
+                mockRegistration);
+
+        when(mockFactory.getTypeName()).thenReturn(TYPE);
+        when(mockFactory.createSubscription(anyObject())).thenReturn(mockSubscription);
+
+        metadata = new SubscriptionMetadata(TYPE, FILTER, CALLBACK);
+        cachedSubscription = new CachedSubscription(metadata) {
+            protected BundleContext getBundleContext() {
+                return mockBundleContext;
+            }
+        };
+    }
+
+    @Test(expected = SubscriptionRegistrationException.class)
+    public void testUnregisterThrowsException() {
+        cachedSubscription.unregisterSubscription();
+    }
+
+    @Test
+    public void testUnregister() {
+        cachedSubscription.registerSubscription(mockSubscription);
+        cachedSubscription.unregisterSubscription();
+        verify(mockRegistration).unregister();
+
+        // Need to ensure the exception happened on the 2nd call to unregister
+        boolean registrationWasSetToNull = false;
+        try {
+            cachedSubscription.unregisterSubscription();
+        } catch (SubscriptionRegistrationException e) {
+            registrationWasSetToNull = true;
+        }
+        assertThat(registrationWasSetToNull, is(true));
```

Comment:
Yeah I don't think we need to be trying to unregister twice in this test. There's already a test for unregistering before registration. I think an assertion on isNotRegistered() is all we need here.